src/applications/disability-benefits/all-claims/utils/submit.js

```diff
@@ -403,6 +409,180 @@ export const addForm0781 = formData => {
+  );
+
+  return { facilityInfo, treatmentMonth, treatmentYear };
+}
+
+/**
+ * Transforms and filters treatment facilities from both provider and VA sources
+ * Only includes facilities where `treatmentLocation0781Related` is true
+ *
+ * @param {array} providerFacilities array of provider facility objects to transform
+ * @param {array} vaFacilities array of VA facility objects to transform
+ * @returns {array} array of transformed facility objects
+ */
+export function transformTreatmentFacilities(
+  providerFacilities = [],
+  vaFacilities = [],
+) {
+  return [
+    ...providerFacilities
+      .filter(f => f.treatmentLocation0781Related)
+      .map(transformProviderFacility),
+    ...vaFacilities
+      .filter(f => f.treatmentLocation0781Related)
+      .map(transformVaFacility),
+  ];
+}
+
+export const addForm0781V2 = formData => {
+  if (!formData.syncModern0781Flow) {
+    return formData;
+  }
+  const clonedData = _.cloneDeep(formData);
+
+  clonedData.form0781 = {
+    ...(clonedData.eventTypes && { eventTypes: clonedData.eventTypes }),
+    ...(clonedData.events && { events: clonedData.events }),
+    ...(clonedData.workBehaviors && {
+      workBehaviors: clonedData.workBehaviors,
+    }),
+    ...(clonedData.healthBehaviors && {
+      healthBehaviors: clonedData.healthBehaviors,
+    }),
+    ...(clonedData.otherBehaviors && {
+      otherBehaviors: clonedData.otherBehaviors,
+    }),
+    ...(clonedData.behaviorsDetails && {
+      behaviorsDetails: clonedData.behaviorsDetails,
+    }),
+    ...(clonedData.supportingEvidenceReports && {
+      supportingEvidenceReports: clonedData.supportingEvidenceReports,
+    }),
+    ...(clonedData.supportingEvidenceRecords && {
+      supportingEvidenceRecords: clonedData.supportingEvidenceRecords,
+    }),
+    ...(clonedData.supportingEvidenceWitness && {
+      supportingEvidenceWitness: clonedData.supportingEvidenceWitness,
+    }),
+    ...(clonedData.supportingEvidenceOther && {
+      supportingEvidenceOther: clonedData.supportingEvidenceOther,
+    }),
+    ...(clonedData.supportingEvidenceUnlisted && {
+      supportingEvidenceUnlisted: clonedData.supportingEvidenceUnlisted,
+    }),
+    ...(clonedData.supportingEvidenceNoneCheckbox && {
+      supportingEvidenceNoneCheckbox: clonedData.supportingEvidenceNoneCheckbox,
+    }),
+    ...(clonedData.treatmentReceivedVaProvider && {
+      treatmentReceivedVaProvider: clonedData.treatmentReceivedVaProvider,
+    }),
+    ...(clonedData.treatmentReceivedNonVaProvider && {
+      treatmentReceivedNonVaProvider: clonedData.treatmentReceivedNonVaProvider,
+    }),
+    ...(clonedData.treatmentNoneCheckbox && {
+      treatmentNoneCheckbox: clonedData.treatmentNoneCheckbox,
+    }),
+    ...(!!clonedData.providerFacility || !!clonedData.vaTreatmentFacilities
+      ? {
+          treatmentProvidersDetails: transformTreatmentFacilities(
+            clonedData.providerFacility || [],
+            clonedData.vaTreatmentFacilities || [],
```

Comment:
I dont think this matters, but is there a difference between how the pdf filler handles a `[]` vs a `nil` or the key just not even being there?

Reply:
In this case, the PDF filler is only going to fill any text values that are present for each facility name and location detail. The result would be the same whether facility types resolved to an empty array or nil value, or if the `treatmentProvidersDetails` key wasn't present.  Regarding the reasoning for defaulting to an empty array, if `clonedData.providerFacility` is `null`, then passing it directly to `transformTreatmentFacilities` would override the default parameter (`[]`), leading to a `TypeError` when calling `.filter()` on `null`. Not sure if any of that answers the question you're asking.